### DISCUSSION TITLE
Can I send metadata or tags to Langfuse?

### DISCUSSION THREAD

**Discussion by giorgosera** [FlowiseAI/Flowise#1648] [Q&A] [answered]
I'd like to be able to pass metadata or at least tags when using Langfuse via Flowise. Is this possible?

Here's what I tried but did not work.

```
{
    "question": "hi there",
    "overrideConfig": {
        "analytics": {
            "langFuse": {
                "userId": "user1",
                "metadata": {
                  "foo": "bar"
                },
                "tags": ["foo"]
            }
        }
    }
}
```

I also tried to find the place in the code where Flowise sets up the trace but I got confused :) It seems that there are multiple places where the setup is happening afaict. Also, please note that I am not using OpenAI.

All the examples, demonstrate how to pass userId but there is not really any documentation on how (and if) this is possible.

Thanks 🙏

**Comment by dkindlund** [comment]
Hey @HenryHengZJ , can you perhaps comment on this?

It seems like the relevant code is located here:
https://github.com/FlowiseAI/Flowise/blob/main/packages/components/src/handler.ts#L270

But I'm also having trouble understanding the right format of the input JSON to get this to work.

**Comment by dkindlund** [comment]
Not sure if this documentation still works but I will try also:
https://langfuse.com/docs/integrations/flowise#override-config

**Comment by dkindlund** [comment]
Okay, so `userId` works:
```
{
  "question": "Hi there.",
  "overrideConfig": {
    "analytics": {
      "langFuse": {
        "userId": "username@domain.com"
      }
    }
  }
}
```

But `tags` does not:
```
{
  "question": "Hi there.",
  "overrideConfig": {
    "analytics": {
      "langFuse": {
        "userId": "username@domain.com",
        "tags": [
          "test",
          "internal"
        ]
      }
    }
  }
}
```

**Comment by dkindlund** [comment]
`version` works:
```
{
  "question": "Hi there.",
  "overrideConfig": {
    "analytics": {
      "langFuse": {
        "userId": "username@domain.com",
        "version": "0.001"
      }
    }
  }
}
```

**Comment by dkindlund** [comment]
And `metadata.tags` does not work:
```
{
  "question": "Hi there.",
  "overrideConfig": {
    "analytics": {
      "langFuse": {
        "userId": "username@domain.com",
        "version": "0.001",
        "metadata": {
          "tags": [
            "fookey",
            "barvalue"
          ]
        }
      }
    }
  }
}
```

**Comment by dkindlund** [comment]
@marcklingen , could you perhaps update your documentation regarding `overrideConfig.analytics.langFuse.version` and also specify whether or not passing through `tags` (or any other metadata) is supported?  If yes, providing an example or two here would be most useful.  Thanks!

**Comment by dkindlund** [comment]
@HenryHengZJ , I guess my question to you here, is what's the right way for a custom tool/code to "pass through" `overrideConfig` data from an initial chatflow and pass it through to a secondary chatflow?  Is the data accessible through the `$flow.overrideConfig` variables -- or no?

**Comment by giorgosera** [comment]
For what is worth in our case we "hijacked" the `version` property and used it to carry our metadata. Obviously, this is not an ideal solution so we'd still be interested in a resolution.

**Comment by marcklingen** [comment]
Sorry for moving slow on this as I was focused on some major new feature releases for Langfuse. I added tags/metadata to the Langchain (JS) integration in langfuse (as of v3.3.4). PR to upgrade to this version in Flowise: https://github.com/FlowiseAI/Flowise/pull/2055/

Thereby passing metdata and tags via the override config to Langfuse should be simple, let me know in case this does not work as expected once the upgrade is released.

**Reply by dkindlund** [reply]
Thanks, @marcklingen -- is there a link to any sort of example documentation showing how to use tags/metadata from within Flowise?  I checked this link: https://langfuse.com/docs/integrations/flowise#override-config but it doesn't seem to be updated.

**Reply by HenryHengZJ** [reply]
@dkindlund @giorgosera above PR was merged. The following should work:
```
{
  "question": "hi there",
  "overrideConfig": {
    "analytics": {
      "langFuse": {
        "userId": "user1",
        "metadata": { "user": "user1@langfuse.com", "tags": ["production"] }
      }
    }
  }
}
```

**Reply by HenryHengZJ** [reply, marked as answer]
<img width="276" alt="image" src="https://github.com/FlowiseAI/Flowise/assets/26460777/68222942-cf58-4d4e-9cde-6b3550df058c">
<img width="431" alt="image" src="https://github.com/FlowiseAI/Flowise/assets/26460777/1bc41d1a-22b4-4ad4-9a47-0ae74cbfab23">

**Reply by dkindlund** [reply]
Thanks, @HenryHengZJ and @marcklingen !
